test/test_rosdep_installers.py

```diff
@@ -614,7 +627,8 @@ def test_RosdepInstaller_install_resolved():
                 raise
             return True
     stdout_lines = [x.strip() for x in stdout.getvalue().split('\n') if x.strip()]
+    sudo_command = 'sudo -H ' if hasattr(os, 'geteuid') and os.geteuid() != 0 else ''
```

Comment:
Since we need to reimplement this logic in many places, I'd suggest that we make it into a utility function that we can test independently and then reuse throughout the implementations and tests.